pkg/management/barman/capabilities/type.go

```diff
@@ -34,3 +38,12 @@ type Capabilities struct {
 	HasAzureManagedIdentity    bool
 	Version                    *semver.Version
 }
+
+// ShouldExecuteBackupWithName returns true if the new backup logic should be executed
+func (c *Capabilities) ShouldExecuteBackupWithName(cluster *apiv1.Cluster) bool {
+	if !c.hasName || cluster == nil {
+		return c.hasName
+	}
```

Comment:
IMO pretty obscure.
we could have `c.hasName` be true, then if `cluster == nil` we return ... true.
Seems wrong, no?